tendrl/commons/jobs/__init__.py

```diff
@@ -56,24 +59,37 @@ def _process_job(self, raw_job, job_key):
     def _acceptor(self):
         while not self.job_consumer_thread._complete.is_set():
             gevent.sleep(2)
-
             # TODO(team) replace below raw write with a "EtcdJobQueue" class
             try:
                 jobs = tendrl_ns.etcd_orm.client.read("/queue")
             except etcd.EtcdKeyNotFound:
                 continue
-                
             for job in jobs.children:
-                executed = False
-                if job.value is None:
+                try:
+                    raw_job = {"job_id": None,
+                               "integration_id": None,
+                               "run": None,
+                               "status": None,
+                               "parameters": None,
+                               "type": None,
+                               "node_ids": None,
+                               "request_id": None,
+                               "errors": None,
+                               "parent": None}
+                    job_id = job.key.split("/")[-1]
+                    result = self.read_job(job.key)
```

Comment:
@anmolbabu ya anmol same,
@Tendrl/tendrl-core
If any fields are empty in class attributes at that time read will never happen for that attribute.
for example we take volume class with attributes vol_id, Vol_name_ vol_size (just for example), There is two machines node A and B. If node A pushing vol_id and vol_name into etcd using volume class. so vol_size is None for node A class. Now node B updating same volume path in etcd with vol_id, vol_name and vol_size (all). Now in etcd volume path contains all fields are populated. If node A try to read volume details from etcd using volume class, then it can be read vol_id and vol_name only, it can't read vol_size. (but vol_size is available in etcd). Problem is some where class attribute values are maintained. so vol_size attribute is None so etcd request for that attribute is not happening.

How this case is affecting jobs means, job is pushed from api, so all the job class attributes are initially empty. so it can't read any value.

how to identify some where value are present ? step: we take disk as example, disk details are saved using save method like Disk(**json).save, so **json contains all disk attribute values. below this save line just give Disk(disk_id="123-12-12-121").save(), here i am giving only disk_id, no other disk details, but if you see in etcd this disk_id present with old disk_id details.